src/rust/plugin-registry/src/server.rs

```diff
@@ -145,17 +224,45 @@ impl PluginRegistryService for PluginRegistry {
     }
 }
 
-pub async fn exec_service(socket_addr: SocketAddr) -> Result<(), Box<dyn std::error::Error>> {
+pub async fn exec_service(
+    service_config: PluginRegistryServiceConfig,
+) -> Result<(), Box<dyn std::error::Error>> {
     let (mut health_reporter, health_service) = tonic_health::server::health_reporter();
     health_reporter
         .set_serving::<PluginRegistryServiceServer<PluginRegistry>>()
         .await;
 
-    let plugin_work_queue = PluginRegistry {};
+    let addr = service_config.plugin_registry_bind_address;
+    tracing::info!(
+        message="Connecting to plugin registry table",
+        plugin_registry_table_username=%service_config.plugin_registry_table_username,
+        plugin_registry_table_hostname=%service_config.plugin_registry_table_hostname,
+        plugin_registry_table_port=%service_config.plugin_registry_table_port,
+    );
+    let postgres_address = format!(
+        "postgresql://{}:{}@{}:{}",
+        service_config.plugin_registry_table_username,
+        service_config.plugin_registry_table_password,
+        service_config.plugin_registry_table_hostname,
+        service_config.plugin_registry_table_port,
+    );
+
+    let plugin_registry: PluginRegistry = PluginRegistry {
+        pool: sqlx::PgPool::connect(&postgres_address)
+            .timeout(std::time::Duration::from_secs(5))
+            .await??,
+        s3: S3Client::from_env(),
+        plugin_bucket_name: service_config.plugin_s3_bucket_name,
+        plugin_bucket_owner_id: service_config.plugin_s3_bucket_aws_account_id,
+    };
+
+    sqlx::migrate!()
+        .run(&plugin_registry.pool)
+        .await?;
 
     tracing::info!(
         message="HealthServer + PluginRegistry listening",
```

Comment:
If there should even be a log message here at all, it should be split into two messages.

The health server started listening all the way at the top of this function (if indeed it started listening; looking at the code, I'm not quite sure that's the case). A lot of other fallible operations happened in the intervening time, and if any of them fail, this message never gets printed.

The PluginRegistry itself hasn't started yet, so a trace implying that it has is misleading.